adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/VertxBasedMqttProtocolAdapter.java

```diff
@@ -251,18 +249,17 @@ private void handleEndpointConnection(final MqttEndpoint endpoint) {
                     });
 
                     // check that MQTT client tries to publish on topic with device_id same as on connection
-                    if (resource.getResourceId().equals(endpoint.clientIdentifier())) {
-
+                    if (getConfig().isSingleTenant() && !resource.getResourceId().equals(endpoint.clientIdentifier())) {
```

Comment:
Must be `if (getConfig().isAuthenticationRequired()` of course (if not being deleted at all).
Will correct it.